src/components/Register/Register.js

```diff
@@ -1,32 +1,46 @@
 import React from 'react';
 import PopupWithForm from '../PopupWithForm/PopupWithForm.js';
+import { useFormWithValidation } from '../FormValidation/FormValidation.js';
 
 function Register(props) {
+    const { values, handleChange, isValid, resetForm } = useFormWithValidation({})
+
+    const submitForm = (e) => {
+        const { email, password, username } = values;
+        e.preventDefault();
+        props.formSubmit(email, password, username);
+        resetForm();
+    }
+
     return (
         <PopupWithForm
-            register={props.register}
+            altFormLink={props.altFormLink}
+            altFormTitle={props.altFormTitle}
             buttonText='Sign up'
-            title='Sign up'
-            type='register'
-            altForm='Sign in'
+            formSubmit={submitForm}
             isOpen={props.isOpen}
-            onClose={props.onClose}
+            isValid={isValid}
             openAltForm={props.openAltForm}
+            onClose={props.onClose}
+            title='Sign up'
+            type='register'
         >
 
             <h6 className='popup__input-title popup__input-title_type_email'>Email</h6>
-            <input required minLength='1' maxLength='30' type='text'
-                className='popup__input popup__input_type_email' placeholder='Email' />
-            <span className='popup__error'>Invalid email address</span>
+            <input minLength='1' maxLength='30' type='email' name='email'
+                className='popup__input popup__input_type_email'
+                placeholder='Email' onChange={handleChange} />
+            <span className={`popup__error ${isValid === false && 'popup__error_visible'}`}>Invalid email address</span>
 
             <h6 className='popup__input-title popup__input-title_type_password'>Password</h6>
-            <input required minLength='1' maxLength='30' type='text'
+            <input minLength='1' maxLength='30' type='text' name='password'
                 className='popup__input popup__input-password'
-                placeholder='Enter password' />
+                placeholder='Enter password' onChange={handleChange} />
 
             <h6 className='popup__input-title popup__input_title_type_username'>Username</h6>
-            <input required minLength='1' maxLength='30' type='text'
-                name='username' className='popup__input popup__input_type_username' placeholder='Username' />
+            <input minLength='1' maxLength='30' type='text' name='username'
+                className='popup__input popup__input_type_username'
+                placeholder='Username' onChange={handleChange} />
```

Comment:
If you have `onChange` in an input then you need to add attribute `value` to this input.
This is how React controls inputs.
To read more about `controlled components`: https://reactjs.org/docs/forms.html#controlled-components